reports/ssl_check/ssl_check.py

```
@@ -0,0 +1,274 @@
+    """
+    def check_certificate_date(self, not_before, not_after, days_cuttoff=30):
+        if not not_after or not not_before:
+            logger.critical("check_date: Missing before or after date")
+            return False
+        # Check the activation date
+        not_before_datetime = datetime.fromtimestamp(not_before)
+        logger.info("check_date: certificate active from {}".format(not_before_datetime.date().isoformat()))
+        if not_before_datetime > datetime.now():
+            logger.critical("check_date: FAILED: certificate not valid until {}".format(not_before_datetime.date().isoformat()))
+            return False
+
+        # Check the expiry date
+        not_after_datetime = datetime.fromtimestamp(not_after)
+        diff_days = (not_after_datetime - datetime.now()).days
+        logger.info("check_date: {} days until expiry ({})".format(diff_days, not_after_datetime))
+        if diff_days < days_cuttoff:
+            logger.critical("check_date: FAILED: {} days until expiry ({})".format(diff_days, not_after_datetime))
+            return False
+
+        return True
+
+    """
+    Check a certificate against our set of passing criteria.
+
+    Args:
+        certificate(dict): Dictionary of certificate data from ssllabs
+
+    Return:
+        (bool): True if the certificate passed our tests, False otherwise
+    """
+    def check_certificate(self, certificate):
+        subject = certificate.get('subject', 'unknown')
+        subject_short = subject.split(',')[0]
+        logger.info("check_certificate: Checking certificate subject '{}'".format(subject_short))
+        # Format the certificates dates from epoch time in microseconds to seconds
+        not_before = certificate.get('notBefore', None) 
+        not_after = certificate.get('notAfter', None)
+        if not_before and not_after and not self.check_certificate_date((not_before * 0.001), (not_after * 0.001)):
```

This line doesn't read super well to me... Maybe don't add the default on the gets above and let the KeyError raise (or catch and log)? Not a deal breaker.